### DISCUSSION TITLE
Using parallel or intercepting routes to render a modal on a catch-all route

### DISCUSSION THREAD

**Discussion by Doesntmeananything** [vercel/next.js#50967] [App Router]
Is it possible to use parallel or intercepting routes when implementing a modal on a catch-all route?

My use case is the following:

1. I have a catch-all route that represents an arbitrary depth in a nested structure (think folders and files in a filesystem).
2. After navigating to this route, I would like my users to be able to open a modal and work within the context of this nested structure, so that the URL looks something like this: `/root/branch-1/leaf-1/edit`, where `/edit` is the last segment used for rendering the modal.

I've looked at the docs for [Parallel routes](https://nextjs.org/docs/app/building-your-application/routing/parallel-routes) and [Intercepting routes](https://nextjs.org/docs/app/building-your-application/routing/intercepting-routes) and tried to add this behaviour to the app, however I get a build error:

```
"Catch-all must be the last part of the URL."
```

Conceptually, I understand why it's not possible to use parallel and intercepting routes for modals like this, since catch-all routes assume they're going to be handling all the route segments at their level and below. But I may also be missing something, e.g. it may be possible to intercept the last segment of a catch-all route and use it to drive modal state.

For now, it seems that my only option for URL-driven modals is to use search parameters.

**Comment by lellimecnar** [comment]
@Doesntmeananything were you able to find a solution?

I resolved it in my project by adding a parallel route as a sibling to the catch-all route, and putting the intercepting route in there. As a side effect, the intercepting route has to apply to everything under the catch-all route, but that's okay in my case. I'm not sure what could be done about that otherwise... Maybe, in the layout, you could conditionally render the parallel route based on the URL? ...Not pretty, but it's working for me.
